math-scala/src/main/scala/org/apache/mahout/math/algorithms/regression/tests/AutocorrelationTests.scala

```diff
@@ -0,0 +1,54 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one
+  * or more contributor license agreements. See the NOTICE file
+  * distributed with this work for additional information
+  * regarding copyright ownership. The ASF licenses this file
+  * to you under the Apache License, Version 2.0 (the
+  * "License"); you may not use this file except in compliance
+  * with the License. You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing,
+  * software distributed under the License is distributed on an
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  * KIND, either express or implied. See the License for the
+  * specific language governing permissions and limitations
+  * under the License.
+  */
+
+package org.apache.mahout.math.algorithms.regression.tests
+
+import org.apache.mahout.math.algorithms.regression.Regressor
+import org.apache.mahout.math.drm.DrmLike
+import org.apache.mahout.math.drm.RLikeDrmOps._
+import org.apache.mahout.math.function.Functions.SQUARE
+import org.apache.mahout.math.scalabindings.RLikeOps._
+
+
+object AutocorrelationTests {
+
+  //https://en.wikipedia.org/wiki/Durbin%E2%80%93Watson_statistic
+  /*
+  To test for positive autocorrelation at significance α, the test statistic d is compared to lower and upper critical values (dL,α and dU,α):
+      If d < dL,α, there is statistical evidence that the error terms are positively autocorrelated.
+      If d > dU,α, there is no statistical evidence that the error terms are positively autocorrelated.
+      If dL,α < d < dU,α, the test is inconclusive.
+
+      Rule of Thumb:
+       d < 2 : positive auto-correlation
+       d = 2 : no auto-correlation
+       d > 2 : negative auto-correlation
+  */
+  def DurbinWatson[K](model: Regressor[K]): Regressor[K] = {
+    val e: DrmLike[K] = model.residuals(1 until model.residuals.nrow.toInt, 0 until 1)
```

Comment:
`nrow.toInt` is generally dangerous as it does not catch the algorithm limitation. The problem with this (and i actually have run into this before) is that the algorithm obviously has in this case a limitation of 2 bln rows, and it should explicitly fall apart once this limit is reached, instead of silently producing a nonsense. I think there's a method specifically for this purpose in one of our math-scala package `drm`,  `safeToNonNegInt` that would throw IllegalArgument if conversion loses significant bits.

Of course the best approach is to avoid such limitations in the first place, but if unavoidable, please use `safeToNonNegInt`.